src/utils/requestProcessingHelper.js

```diff
@@ -0,0 +1,26 @@
+import { requestStates } from "./../constants";
+
+export const computeFinalRequestProcessingState = function () {
+	const states = arguments;
+	let idleCount = 0;
+
+	for (let i = 0; i < states.length; i++) {
+		switch (states[i]) {
+			case requestStates.fail:
+				return requestStates.fail;
+			case requestStates.processing:
+				return requestStates.processing;
+			case requestStates.idle:
+				idleCount++;
+				continue;
+			case requestStates.success:
+				continue;
+			default:
+				throw Error("You have to pass just requestStates type");
```

Comment:
Only valid requestStates can be used

Reply:
Is there really big difference between those 2 statements? 😄